test-jest/component/Text.spec.tsx

```diff
@@ -0,0 +1,216 @@
+
+    setTimeout(() => {
+      expect(text.textContent).toContain('0');
+    }, 1000);
+  });
+
+  test('Render text when x or y is a percentage', () => {
+    render(
+      <Text role="img" x="50%" y="50%">
+        anything
+      </Text>,
+    );
+
+    const text = screen.getByRole('img');
+    expect(text).toBeInTheDocument();
+
+    setTimeout(() => {
+      expect(text.textContent).toContain('anything');
+    }, 1000);
+  });
+
+  test("Don't Render text when x or y is NaN", () => {
+    render(
+      <Text role="img" x={NaN} y={10}>
+        anything
+      </Text>,
+    );
+
+    const text = screen.queryByRole('img');
+    expect(text).not.toBeInTheDocument();
+  });
+
+  test('Only split contents on breaking spaces', () => {
+    const testString = 'These spaces\tshould\nbreak,\rbut\xA0these\xA0should\xA0not.';
+    render(
+      <Text role="img" width="auto">
+        {testString}
+      </Text>,
+    );
+
+    const text = screen.getByRole('img');
+    expect(text).toBeInTheDocument();
+
+    expect(text?.children).toHaveLength(5);
+  });
+
+  describe('maxLines', () => {
+    test('does not do anything when maxLines are not exceeded', () => {
+      render(
+        <Text role="img" width={500} maxLines={3}>
+          test
+        </Text>,
+      );
+      render(
+        <Text role="img" width={500}>
+          test
+        </Text>,
+      );
+
+      const text = screen.getAllByRole('img');
+      expect(text[0]).toBeInTheDocument();
+
+      expect(text[0]?.textContent).toEqual(text[1]?.textContent);
+    });
+
+    test('limits the output to maxLines', () => {
+      const testString = `Lorem ratione omnis fuga dignissimos in amet. Minus quam architecto non ea iste!
+        Nihil amet in itaque error velit. Corporis autem sequi aut temporibus placeat.
+        Perferendis quos veritatis quasi pariatur!`;
+      render(
+        <Text role="img" width={200} maxLines={2}>
+          {testString}
+        </Text>,
+      );
+
+      const text = screen.getByRole('img');
+      expect(text).toBeInTheDocument();
+
+      expect(text?.children).toHaveLength(2);
+    });
+
+    test('adds an ellipsis at the end of the truncated line', () => {
+      const testString = `Sit totam suscipit aliquid suscipit eius, cupiditate Aut excepturi ipsum ut suscipit
+        facilis debitis Provident impedit a distinctio neque quaerat Optio quo quibusdam possimus
+        provident accusantium. Molestiae similique nemo labore`;
+      render(
+        <Text role="img" width={200} maxLines={2}>
+          {testString}
+        </Text>,
+      );
+
+      const text = screen.getByRole('img');
+      expect(text).toBeInTheDocument();
+      const lastChild = text?.children[text?.children?.length - 1];
+      const lastLetter = lastChild.textContent?.[lastChild?.textContent?.length - 1];
+
+      expect(lastLetter).toEqual('…');
+    });
+
+    // test('adds an ellipsis at the end of a very long word', () => {
```

Comment:
@nikolasrieble @Yilun-Sun @proke03 any ideas on how to get this to work?

As you can see above I mock `getBoundingClientRect` to specific widths - the problem with that is `Text` has a lot of logic relying on `getStringSize` - apparently `maxLines={1}` with a long string is supposed to result in ellipses but I can't get it to work as it should because the strings can't be measured properly.

Something in here https://github.com/recharts/recharts/blob/master/src/component/Text.tsx#L33

as well as where its called https://github.com/recharts/recharts/blob/master/src/component/Text.tsx#L130-L155
(`originalResult` is always returned here which doesn't have ellipsis)

Reply:
Well, you could mock getStringSize as well, and then we could separately unit test getStringSize.

Reply:
Although Text has several logics inside, I don't think it is necessary to test the output of text according to the values of those logics. You can write tests according to the applicable case.

Reply:
We already separately test `getStringSize` (we can't really test it though, we just have to assert it gives back the mocked value). 

The same issue will occur if we mock `getStringSize` instead of `getBoundingClientRect` - there is a loop that processes the characters in the string. We would need to change the value on every loop iteration which I think is impossible.

Reply:
I am fine with creating a separate issues to track this test, and unblock the migration. Same with the two missing tests for the xAxis in your PR @proke03